app/products/calls/actions/calls.ts

```diff
@@ -236,3 +248,96 @@ export const setSpeakerphoneOn = (speakerphoneOn: boolean) => {
     InCallManager.setSpeakerphoneOn(speakerphoneOn);
     setSpeakerPhone(speakerphoneOn);
 };
+
+export const endCallAlert = async (serverUrl: string, intl: IntlShape, channelId: string) => {
```

Comment:
I wonder if the action should be endCall and all the checks and alerts should be handled in the component instead if here.

I'd be ok with a local action that does the checks and the remote action that ends the call while everything Alert /localization related happens in the component.

What do you think?

Reply:
@enahum For sure we could do that -- but then there would be a lot of calls-specific code in the `post_draft/send_handler`, and I thought that would be worse. I agree this function is not very elegant, but at least it keeps the mess confined to the function and lets the `send_handler` handle its own responsibilities.
Unless you had a different solution? I'm open to whatever you think is best.

Reply:
@cpoile I totally understand your reasoning, and it makes a lot of sense from one POV.

That said, and we can have this conversation over a "Call" if you think is best, in my opinion, Calls is here to stay and even though is another product, in case of the mobile client is actually part of the core codebase, so I don't see any issues with having it less isolated, the `/call` command is invoking this action from within the `send_handler` component anyway. So this is another POV.

Anyway, I just thought that it would've made more sense, but in the end is what your team deems to be best, as most probably is going to be easier to debug and maintain if is done the way is implemented. You call (pun intended :P)

Reply:
@enahum I think this is better -- it keeps the alerts and user-facing code in `SendHandler`, but puts the state handling in specific functions in Calls files. It means we don't have to introduce a lot of very specific state into `SendHandler` for a slash command that will be run very infrequently (as we discussed). Let me know what you think?